src/System.Text.Json/tests/Serialization/EnumTests.cs

```diff
@@ -385,14 +385,93 @@ public static void EnumAsUInt16Max()
         public static void EnumAsByteMax()
         {
             SimpleTestClass obj = JsonSerializer.Parse<SimpleTestClass>(s_jsonByteEnum);
-            Assert.Equal(SampleByteEnum.Max, obj.MyByteEnum);
+            Assert.Equal(SampleEnumByte.Max, obj.MyByteEnum);
         }
 
         [Fact]
         public static void EnumAsSByteMax()
         {
             SimpleTestClass obj = JsonSerializer.Parse<SimpleTestClass>(s_jsonSByteEnum);
-            Assert.Equal(SampleSByteEnum.Max, obj.MySByteEnum);
+            Assert.Equal(SampleEnumSByte.Max, obj.MySByteEnum);
+        }
+
+        [Theory]
+        [InlineData(sbyte.MinValue, SampleEnumSByte.MinNegative)]
+        [InlineData(short.MinValue, SampleEnumInt16.MinNegative)]
+        [InlineData(int.MinValue, SampleEnumInt32.MinNegative)]
+        [InlineData(long.MinValue, SampleEnumInt64.MinNegative)]
+        [InlineData(0, SampleEnum.MinZero)]
+        [InlineData(0, SampleEnumByte.MinZero)]
+        [InlineData(0, SampleEnumUInt16.MinZero)]
+        [InlineData(0, SampleEnumUInt32.MinZero)]
+        [InlineData(0, SampleEnumUInt64.MinZero)]
+        [InlineData(0, SampleEnumSByte.Zero)]
+        [InlineData(0, SampleEnumInt16.Zero)]
+        [InlineData(0, SampleEnumInt32.Zero)]
+        [InlineData(0, SampleEnumInt64.Zero)]
+        [InlineData(1, SampleEnum.One)]
+        [InlineData(1, SampleEnumByte.One)]
+        [InlineData(1, SampleEnumSByte.One)]
+        [InlineData(1, SampleEnumInt16.One)]
+        [InlineData(1, SampleEnumUInt16.One)]
+        [InlineData(1, SampleEnumInt32.One)]
+        [InlineData(1, SampleEnumUInt32.One)]
+        [InlineData(1, SampleEnumInt64.One)]
+        [InlineData(1, SampleEnumUInt64.One)]
+        [InlineData(2, SampleEnum.Two)]
+        [InlineData(2, SampleEnumByte.Two)]
+        [InlineData(2, SampleEnumSByte.Two)]
+        [InlineData(2, SampleEnumInt16.Two)]
+        [InlineData(2, SampleEnumUInt16.Two)]
+        [InlineData(2, SampleEnumInt32.Two)]
+        [InlineData(2, SampleEnumUInt32.Two)]
+        [InlineData(2, SampleEnumInt64.Two)]
+        [InlineData(2, SampleEnumUInt64.Two)]
+        [InlineData(byte.MaxValue, SampleEnumByte.Max)]
+        [InlineData(sbyte.MaxValue, SampleEnumSByte.Max)]
+        [InlineData(short.MaxValue, SampleEnumInt16.Max)]
+        [InlineData(ushort.MaxValue, SampleEnumUInt16.Max)]
+        [InlineData(int.MaxValue, SampleEnumInt32.Max)]
+        [InlineData(uint.MaxValue, SampleEnumUInt32.Max)]
+        [InlineData(long.MaxValue, SampleEnumInt64.Max)]
+        [InlineData(ulong.MaxValue, SampleEnumUInt64.Max)]
+        public static void ToString_Max(object expected, object enumValue)
+        {
+            string json = JsonSerializer.ToString(enumValue);
+            Assert.Equal(expected.ToString(), json);
+        }
+
+        [Theory]
+        [InlineData(1, (long)byte.MaxValue + 1, typeof(SampleEnumByte), SampleEnumByte.MinZero)]
```

Comment:
**Question:** Why are we doing `+ 1` for the "maxCapacity"?

Reply:
Because max value is MaxValue (255 for byte) but including 0, we have is `MaxValue + 1` numbers that can be represented by byte (0..255), so 256 would be the maxCapacity

Reply:
A comment within the test with that explanation would be useful for any reviewer.

Reply:
sure, will add the note.